internal/controller/instaslice_controller.go

```diff
@@ -682,16 +688,17 @@ func (r *InstasliceReconciler) setInstasliceAllocationToDeleting(ctx context.Con
 	var updateInstasliceObject inferencev1alpha1.Instaslice
 	typeNamespacedName := types.NamespacedName{
 		Name:      instasliceName,
-		Namespace: InstaSliceOperatorNamespace, // TODO: modify if needed
+		Namespace: InstaSliceOperatorNamespace,
 	}
 	errRetrievingInstaSlice := r.Get(ctx, typeNamespacedName, &updateInstasliceObject)
 	if errRetrievingInstaSlice != nil {
 		log.Error(errRetrievingInstaSlice, "error getting latest instaslice object")
 		return ctrl.Result{Requeue: true}, errRetrievingInstaSlice
 	}
-
+	original := updateInstasliceObject.DeepCopy()
 	updateInstasliceObject.Spec.Allocations[podUUID] = allocation
-	errUpdatingInstaslice := r.Update(ctx, &updateInstasliceObject)
+	patch := client.MergeFrom(original)
+	errUpdatingInstaslice := r.Patch(ctx, &updateInstasliceObject, patch)
```

Comment:
stick with golang convention for naming the variables.

Reply:
renamed to `err`